src/Components/animation/Animation.js

```diff
@@ -137,377 +69,265 @@ class Animation extends React.Component {
-      toValue: 0,
-      duration: 0,
-      easing: Easing.linear,
-    }).start();
-  }
-
-  /** Play the whole animation */
-  _restart() {
-    this._initPositions();
-
-    /* Animation of the whole animation */
-    var completeSequence = [];
-
-    completeSequence.push(
-      Animated.timing(this.currentStepAV, {
-        toValue: 0,
-        duration: 0,
-        easing: Easing.linear,
-      }),
-    );
-
-    /* For each step (start at step 1 as step 0 corresponds to the initial positions)*/
-    for (var stepId = 1; stepId < this._stepCount(); stepId++)
-      completeSequence.push(Animated.parallel(this._getStepAnimation(stepId, true)));
-
-    Animated.sequence(completeSequence).start();
-
-    if (this.props.onStepChange !== undefined && this.props.onStepChange !== null)
-      this.props.onStepChange(this._stepCount());
   }
 
   /** Get back to the previous step */
   _previousStep() {
     if (this.state.currentStep !== 0) {
-      this.setState(
-        prevState => ({ currentStep: prevState.currentStep - 1 }),
-        () => {
-          var stepArray = this._getStepAnimation(this.state.currentStep, false);
-          Animated.parallel(stepArray).start();
-
-          if (this.props.onStepChange !== undefined && this.props.onStepChange !== null)
-            this.props.onStepChange(this.state.currentStep);
-        },
-      );
+      this.moveToStep(this.state.currentStep - 1);
+
+      if (this.props.onStepChange !== undefined && this.props.onStepChange !== null)
+        this.props.onStepChange(this.state.currentStep);
     }
   }
 
   /** Go to the next step */
   _nextStep() {
     if (this.state.currentStep !== this._stepCount() - 1) {
-      this.setState(
-        prevState => ({ currentStep: prevState.currentStep + 1 }),
-        () => {
-          Animated.parallel(this._getStepAnimation(this.state.currentStep, true)).start();
-
-          if (this.props.onStepChange !== undefined && this.props.onStepChange !== null)
-            this.props.onStepChange(this.state.currentStep);
-        },
-      );
+      this.playStep(this.state.currentStep + 1);
+
+      if (this.props.onStepChange !== undefined && this.props.onStepChange !== null)
+        this.props.onStepChange(this.state.currentStep);
     }
   }
 
-  /** Returns the animation to a given step for all displayed elements
-    The substeps are played if the animation is forward and if the elements moves at step stepId
-  */
-  _getStepAnimation = (stepId, isForward) => {
-    debug('Animation: get animation: ' + stepId);
-
-    stepId = Math.max(0, Math.round(stepId));
-
-    /* Animation of all the elements at step stepId */
-    var stepAnimation = [];
-
-    /* The first animation in the sequence enables to update the current step */
-    stepAnimation.push(
-      Animated.timing(this.currentStepAV, {
-        toValue: stepId,
-        duration: this.state.stepLength,
-        easing: Easing.linear,
-        key: 0,
-      }),
+  goToStep = stepId => {
```

Comment:
Do you think we really need this function, or we could remove it and use directly `this.playStep` where it's called?

Reply:
Yes we can. Initially, goToStep was:
- playing a step if the step was after the current step; and 
- moving immediately to a step if the step was before the current step;
but it was confusing.

Reply:
ah ok